python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -471,21 +471,47 @@ def __append_optimize_op__(op, block, grad_to_block_id, merged_var):
                 self._append_pserver_ops(block, op, endpoint, grad_to_block_id,
                                          self.origin_program, merged_var)
             else:
-                self._append_pserver_non_opt_ops(block, op, endpoint)
+                self._append_pserver_non_opt_ops(block, op)
 
         def __op_have_grad_input__(op):
             for varname in op.input_arg_names:
                 if varname.find("@GRAD") >= 0:
                     return varname
             return ""
 
+        def __clone_lr_op_sub_block__(op, program, new_block):
+            if not op.has_attr('sub_block'):
+                return
+
+            origin_block_desc = op.attr('sub_block')
+            origin_block = self.origin_program.block(origin_block_desc.id)
+            assert isinstance(origin_block, Block)
+            # we put the new sub block to new block to follow the block
+            # hierarchy of the original blocks
+            new_sub_block = program.create_block(new_block.idx)
```

Comment:
Adding blocks to `pserver_program` will corrupt the execution of parameter server, for the sub-blocks may be executed automatically by the `listen_and_serv` op. This PR should be reverted.